packages/components/src/tour_tip/tour_tip.tsx

```diff
@@ -100,6 +103,7 @@ export const TourTip = ({
         if (!showBackdrop && handleDismiss) {
             handleDismiss(e);
         }
+        handleInteractivePunchOut();
```

Comment:
Making the default value of `handleInteractivePunchOut` == `noop` so that we don't have to add another if statement.

Reply:
@MattSilvaa

Reply:
@MattSilvaa Instead of introducing another function, I think we can handle it internally with something like this
<img width="853" alt="Screenshot 2023-02-06 at 6 51 59 PM" src="https://user-images.githubusercontent.com/16203333/216982395-64609755-f855-4661-9527-01308b7a22de.png">
<img width="289" alt="Screenshot 2023-02-06 at 6 53 48 PM" src="https://user-images.githubusercontent.com/16203333/216982651-2ee1ed6a-6aab-4218-a240-71c160fb6f3d.png">

Reply:
@mattermod, what do you think?

Reply:
@michelengelen, what do you think?

Reply:
this is generally a good idea, but you normally would not need to have the `useLayoutEffect`, since a change in props would re-render the component anyways and set a clean state.

and you can use optional parameter instead of the `if` guards as well:

```
handleJump?.(event, jumpToStep)
```

and
 
```
handleDismiss?.(e)
```

to make it cleaner.

Reply:
oh and the convention for state notation is `const [<something>, set<Something>] = useState()`

Reply:
Thanks for the suggestions @AshishDhama & @michelengelen, I have pushed up the feedback 👍🏻